src/H5back_gen.py

```diff
@@ -0,0 +1,472 @@
+           'VL_VECTOR_INT': vl_reader,
+           'VECTOR_DOUBLE': vector_reader,
+           'VL_VECTOR_DOUBLE': vl_reader,
+           'VECTOR_FLOAT': vector_reader,
+           'VL_VECTOR_FLOAT': vl_reader,
+           'VECTOR_STRING': vector_string_reader,
+           'VECTOR_VL_STRING': vector_vl_string_reader,
+           'VL_VECTOR_STRING': vl_reader,
+           'VL_VECTOR_VL_STRING': vl_reader,
+           'SET_INT': set_list_x_reader,
+           'VL_SET_INT': vl_reader,
+           'SET_STRING': set_string_reader,
+           'VL_SET_STRING': vl_reader,
+           'SET_VL_STRING': set_vl_string_reader,
+           'VL_SET_VL_STRING': vl_reader,
+           'LIST_INT': set_list_x_reader,
+           'VL_LIST_INT': vl_reader,
+           'LIST_STRING': list_string_reader,
+           'VL_LIST_STRING': vl_reader,
+           'VL_LIST_VL_STRING': vl_reader,
+           'LIST_VL_STRING': list_vl_string_reader,
+           'PAIR_INT_INT': pair_int_int_reader,
+           'PAIR_INT_STRING': pair_int_string_reader,
+           'PAIR_INT_VL_STRING': pair_int_vl_string_reader,
+           'MAP_INT_INT': map_xy_reader,
+           'VL_MAP_INT_INT': vl_reader,
+           'MAP_INT_DOUBLE': map_xy_reader,
+           'VL_MAP_INT_DOUBLE': vl_reader,
+           'MAP_INT_STRING': map_int_string_reader,
+           'VL_MAP_INT_STRING': vl_reader,
+           'MAP_INT_VL_STRING': map_int_vl_string_reader,
+           'VL_MAP_INT_VL_STRING': vl_reader,
+           'MAP_STRING_INT': map_string_x_reader,
+           'VL_MAP_STRING_INT': vl_reader,
+           'MAP_VL_STRING_INT': map_vl_string_x_reader,
+           'VL_MAP_VL_STRING_INT': vl_reader,
+           'MAP_STRING_DOUBLE': map_string_x_reader,
+           'VL_MAP_STRING_DOUBLE': vl_reader,
+           'MAP_STRING_STRING': map_string_string_reader,
+           'VL_MAP_STRING_STRING': vl_reader,
+           'MAP_STRING_VL_STRING': map_string_vl_string_reader,
+           'VL_MAP_STRING_VL_STRING': vl_reader,
+           'MAP_VL_STRING_DOUBLE': map_vl_string_x_reader,
+           'VL_MAP_VL_STRING_DOUBLE': vl_reader,
+           'MAP_VL_STRING_STRING': map_vl_string_string_reader,
+           'VL_MAP_VL_STRING_STRING': vl_reader,
+           'MAP_VL_STRING_VL_STRING': map_vl_string_vl_string_reader,
+           'VL_MAP_VL_STRING_VL_STRING': vl_reader,
+           'MAP_PAIR_INT_STRING_DOUBLE': map_pair_int_string_double_reader,
+           'VL_MAP_PAIR_INT_STRING_DOUBLE': vl_reader,
+           'MAP_PAIR_INT_VL_STRING_DOUBLE': map_pair_int_vl_string_double_reader,
+           'VL_MAP_PAIR_INT_VL_STRING_DOUBLE': vl_reader}
+
+query_cases = ''
+for db in dbtest: 
+    current_type = TypeStr(db)
+    reader = readers[db]
+    ctx = {"t": current_type,
+           "no_close": no_close,
+           "H5Tclose": H5Tclose,
+           "H5Tclose_multi": H5Tclose_multi}
+    query_cases += case_template.format(t = current_type, read_x = textwrap.indent(reader.format(**ctx), indent))
+
+print(query_cases)
```

Comment:
This block should be in a `main()` function.  Then the following code should be added to the file:

```python
if __name__ == '__main__':
    main()
```